src/docker-osx-dev

```diff
@@ -389,7 +389,8 @@ function init_boot2docker {
 # Inspect especific configuration about the docker-machine
 #
 function inspect_docker_machine {
-  docker-machine inspect --format="$*" "$DOCKER_MACHINE_NAME" 2>&1
+  local result=$(docker-machine inspect --format="$*" "$DOCKER_MACHINE_NAME" 2>&1)
+  echo ${result/<no value>/}
```

Comment:
Yes, you have it right. I think this is a one or the other situation. My bash is pretty rubbish so maybe there's some other way to do this that takes one and falls back to the other.

Detail:

When the value is not there, it returns `<no value>` on 0.5.0 (which in this case is for `{{.StorePath}}`. When i ran it against 0.4.1 I didn't check that specifically, but that is the idea.

Based on the code I posted from docker-machine master, I believe this field is migrated from one location to another, however downgrading from 0.5.0 to 0.4.1 messes up this particular field, so I'm quite reluctant to try again (takes me a few hours to rebuild my images).

Don't know about neither, but then the solution can't work anyway...

I don't think both would ever appear, again based on the object definitions in the code base. Versions 0 and 1 of Host have them in the old place, and Version 2 has it in the AuthOptions and does not copy it to the old location.

That being said, all this is an undocumented data structure, so hard to speak in certainties.